content/blog/tidymodels-2023-q2/index.Rmd

```diff
@@ -0,0 +1,173 @@
+
+ames
+```
+
+The new datasets are:
+
+* `data_animals()` contains a long-form description of the animal (in the `text` column) as well as quite a bit of missing data and malformed fields.
+* `data_chimiometrie_2019()` contains spectra measured at 550 (unknown) wavelengths, published as the challenge at the Chimiometrie 2019 conference.
+* `data_elevators()` contains information on a subset of the elevators in New York City. 
+
+Because those datasets are stored online, accessing them requires an active internet connection. We plan on using those datasets mostly for workshops and websites. The datasets in the modeldata package are part of the package directly, so they can be used everywhere (regardless of an active internet connection). We typically use them for package documentation.
+
+
+## spatialsample
+
+spatialsample is a package for spatial resampling, extending the rsample framework to help create spatial extrapolation between your analysis and assessment data sets.
+
+The latest release of spatialsample includes nearest neighbor distance matching (NNDM) cross-validation via `spatial_nndm_cv()`. NNDM is a variant of leave-one-out cross-validation which assigns each observation to a single assessment fold, and then attempts to remove data from each analysis fold until the nearest neighbor distance distribution between assessment and analysis folds matches the nearest neighbor distance distribution between training data and the locations a model will be used to predict. [Proposed by Milà et al. (2022)](https://doi.org/10.1111/2041-210X.13851), this method aims to provide accurate estimates of how well models will perform in the locations they will actually be predicting. This method was originally implemented in the CAST package and can now be used with spatialsample as well.
+
+Let's use the Ames housing data and turn it from a regular tibble into a `sf` object for spatial data.
+
+```{r}
+library(spatialsample)
+
+ames_sf <- sf::st_as_sf(ames, coords = c("longitude", "latitude"), crs = 4326)
+```
+
+Let's assume that we are building a model to predict observations similar to this subset of the data:
+
+```{r}
+ames_prediction_sites <- ames_sf[2001:2100, ]
+```
+
+Let's create NNDM cross-validation folds from a reduced training set as an example, just to keep things light.
+
+```{r}
+ames_folds <- spatial_nndm_cv(ames_sf[1:100, ], ames_prediction_sites)
+```
+
+The resulting `rset` contains 100 splits of the data, always keeping 1 of the 100 data points in the assessment set.
+
+```{r}
+ames_folds
+```
+
+Starting with all other 99 points in the analysis set, points are excluded until the distribution of nearest neighbor distances from the analysis set to the assessment set matches that of nearest neighbor distances from the training set to the prediction sites.
+
+Looking at one of the splits, we can see the single assessment point, the points included in the analysis set, and the points excluded as the buffer.
+
+```{r}
```

Comment:
Totally uptight of me but... can you make the physical size of each axis the same (square figure region)?

Reply:
Hm, I've played around with this but I'd like to show a plain `autoplot()` without tweaks here.

In more detail: These seem like sensible defaults to me, I don't quite see why the data would ask for a square layout, a square layout is a little hard to land on with this coordinate system, and I don't want to divert attention from the content of the plot and the out-of-the-box syntax with the customization.